packages/cozy-client/src/CozyClient.js

```diff
@@ -309,21 +310,26 @@ class CozyClient {
    *
    * Warning: unlike other instantiators, this one needs to be awaited.
    *
-   * @param {OldCozyClient} oldClient - An instance of the deprecated cozy-client
+   * @param {OldCozyClient} oldClient - An OAuth instance of the deprecated cozy-client
+   * @param {object} options - CozyStackClient options
    * @returns {Promise<CozyClient>} An instance of a client, configured from the old client
    */
   static async fromOldOAuthClient(oldClient, options) {
-    const hasOauthCreds = oldClient._oauth && oldClient._authcreds != null
-    if (hasOauthCreds) {
-      const token = (await oldClient._authcreds).token
+    if (oldClient._oauth) {
+      const credentials = await oldClient.authorize()
+      const oauthOptions = {
+        oauth: credentials.client,
+        token: credentials.token,
+        scope: credentials.token.scope,
+        ...options
```

Comment:
nit:
Why did you move `...option` into `oauthOptions`?

Result is the same as when put in `CozyClient` constructor, but here it seems like `options` should only contain oauth related properties. Is that wanted?